### DISCUSSION TITLE
Content Collections Metadata

### DISCUSSION THREAD

**Discussion by lorenzolewis** [withastro/roadmap#425] [Proposal]
Right now there isn't a build-in way (that I've discovered at least) to add additional metadata for a content collection.

Let's say that I have this collection:

```ts
const blog = defineCollection({
  schema: {
    date: z.date(),
    title: z.string(),
    summary: z.string().optional(),
    image: z.string().optional(),
    authors: z.array(z.string()).optional(),
  },
})
```

I'd like to be able to an item here that contains the human-friendly version of this collection (`title: "Blog"` for a very basic example).

I could see this being achieved with something similar to this:
```ts
const blog = defineCollection({
  // new "meta" object below
  meta: {
    title: "Blog"
  },
  schema: {
    date: z.date(),
    title: z.string(),
    summary: z.string().optional(),
    image: z.string().optional(),
    authors: z.array(z.string()).optional(),
  },
})
```

**Comment by florian-lefebvre** [comment]
What would be the usecase?

**Comment by lloydjatkinson** [comment]
If it helps this is how I "add" data to content collections, by inferring and deriving a new type. Astro doesn't fill in these new values, I do, usually after `.map()`-ing over `getCollection()` in `getStaticPaths()`. This also means I generally need to pass `Article[]` types around my codebase rather than `ContentCollection[xxx]`.

```ts
export const articleSchema = ({ image }: SchemaContext) =>
    z.object({
        title: z.string().transform((title) => apStyleTitleCase(title)),
        description: z.string().optional(),
        categories: z.array(z.string()),
        dates: datesSchema,
        images: imagesSchema({ image }).optional(),
        related: relatedSchema.optional(),
        status: statusSchema.optional(),
    });
export type Article = Readonly<
    z.infer<ReturnType<typeof articleSchema>> & {
        id: string;
        previous?: Article;
        next?: Article;
    }
>;
```
